icu4c/source/test/intltest/intltest.cpp

```diff
@@ -2143,18 +2146,29 @@ UBool IntlTest::assertEquals(const char* message,
     return true;
 }
 
-UBool IntlTest::assertEquals(const char* message,
-                             UBool expected,
-                             UBool actual) {
+bool IntlTest::assertBooleanEquals(const char *message, int8_t expected, int8_t actual) {
     if (expected != actual) {
-        errln(UnicodeString("FAIL: ") + message + "; got " +
-              toString(actual) +
-              "; expected " + toString(expected));
+        errln(UnicodeString("FAIL: ") + message + "; got " + toString(actual) + "; expected " +
+              toString(expected));
         return false;
     }
 #ifdef VERBOSE_ASSERTIONS
     else {
-      logln(UnicodeString("Ok: ") + message + "; got " + toString(actual));
+        logln(UnicodeString("Ok: ") + message + "; got " + toString(actual));
+    }
+#endif
+    return true;
+}
+
+bool IntlTest::assertBooleanNotEquals(const char *message, int8_t expected, int8_t actual) {
```

Comment:
for ...NotEquals(), should we share implementations with the positive versions and parameterize?

Reply:
Doesn’t seem to be worth it for now. If we want to factorize more, I would probably go instead in the direction of a single assertEquals and a single assertNotEquals implementation, with a debugString function to show the values.